gateway-js/src/supergraphManagers/LegacyFetcher/index.ts

```diff
@@ -0,0 +1,259 @@
+// TODO(trevor:removeServiceList) the whole file goes away
+import { Logger } from 'apollo-server-types';
+import resolvable from '@josephg/resolvable';
+import {
+  SupergraphManager,
+  SupergraphSdlHookOptions,
+  DynamicGatewayConfig,
+  isSupergraphSdlUpdate,
+  isServiceDefinitionUpdate,
+  ServiceDefinitionUpdate,
+  GetDataSourceFunction,
+} from '../../config';
+import {
+  Experimental_UpdateComposition,
+  SubgraphHealthCheckFunction,
+  SupergraphSdlUpdateFunction,
+} from '../..';
+import { composeAndValidate, compositionHasErrors, ServiceDefinition } from '@apollo/federation';
+import { GraphQLSchema, isIntrospectionType, isObjectType, parse } from 'graphql';
+import { buildComposedSchema } from '@apollo/query-planner';
+import { defaultFieldResolverWithAliasSupport } from '../../executeQueryPlan';
+
+export interface LegacyFetcherOptions {
+  pollIntervalInMs?: number;
+  logger?: Logger;
+  subgraphHealthCheck?: boolean;
+  updateServiceDefinitions: Experimental_UpdateComposition;
+  gatewayConfig: DynamicGatewayConfig;
+}
+
+type State =
+  | { phase: 'initialized' }
+  | { phase: 'polling'; pollingPromise?: Promise<void> }
+  | { phase: 'stopped' };
+
+export class LegacyFetcher implements SupergraphManager {
```

Comment:
I don't understand the difference between LegacyFetcher and IntrospectAndCompose (given that the latter is the drop-in replacement for serviceList and the former is how serviceList is implemented).  The files also seem to have a lot in common.

I think it's pretty reasonable to have the similar files instead of stressing about factoring out common code (esp if you're deleting this in a few versions), but a comment explaining how this differs from IAC would really help.

Reply:
Added comments at the top of the file. The goal here was to make the removal of the experimental update options easy, and soon. I could mix this into `IntrospectAndCompose` but it'll result in handling different cases/types of config and have to be ripped out / de-intertwined in the near future.

Reply:
Aha, I missed that IntrospectAndCompose *is* actually being used for serviceList and this really is only for those extra hooks too.

The code duplication is not great if we don't follow through on the removal but sounds like we will, so this is good!